src/structures/WelcomeChannel.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const Base = require('./Base');
+const Emoji = require('./Emoji');
+
+/**
+ * Represents a channel link in a guild's welcome screen.
+ */
+class WelcomeChannel extends Base {
+  constructor(guild, data) {
+    super(guild.client);
+    /**
+     * The guild for this welcome channel
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    this._patch(data);
+  }
+
+  /**
+   * Builds the welcome channel with the provided data.
+   * @param {*} data The raw data of this welcome channel
+   * @private
+   */
+  _patch(data) {
+    if (!data) return;
```

Comment:
```suggestion

```
Is there ever a case where the library attempts to construct this without data?

Reply:
same as above, if discord decides to send updates to this there might be instances where patch is called, but no data is present